### PR TITLE
Postgres support

### PR DESCRIPTION
**Pull request by kajuberdut**
Added PostgreSQL support through SQL Alchemy. All tests are currently passing and both the current app and api server run without any obvious errors. This also adds a  config file that will default to debugging on and in-memory SQLite db if no config is found.

### CODE REVIEW

**Comment by kajuberdut** [comment]
A note, we should possibly squash all my comments if we merge.
Also, I had to rebase to get the Travis CI items and am not 100% sure I did that well. Would appreciate comments if artifacts of rebasing are found.

File: db/utils.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            if not os.path.isfile(path):
+                print("No config.json found, creating default config.")
+                with open(path, 'w') as conf:
+                    conf.write(json.dumps(self.defaultconfig))
+            with open(path, 'r') as conf:
+                self.config = json.loads(conf.read())
+        
+        def getconfig(self, handle):
+                return self.config.get(handle)
+
```
Comment (mtik00) [inline db/utils.py]: For bonus points, look up `__getitem__`!

Comment (kajuberdut) [reply db/utils.py]: Nifty. Only, now I feel like I should just be subclassing dict to make this config class. Nothing wrong with that I guess.

Comment (kajuberdut) [reply db/utils.py]: Actually... it turns out the standard library has a wonderful configparser. Because of course it does.

**Comment by kajuberdut** [comment]
in Travis CI: py.test is passing, then the lint report is showing up, finally everything is failing with the error:

> make: *** [lint] Error 4

**Pull request merged**